doc/0003-wizarding-api.md

```diff
@@ -0,0 +1,227 @@
+> It creates a tight coupling with a specific wizard or editor and it cannot be
+> guaranteed that your expectation about the `stuff` will be met!
+
+```ts
+type WizardRequest = {
+	stuff?: unknown
+}
+```
+
+> **ℹ️ INFO:** you can find the up-to-date definitions here: TODO: file link
+
+> **ℹ️ INFO**: More about other alternatives of requests: 
+> [Alternatives for Requests](#alternatives-of-requests)
+
+##### Finding Available Wizards
+
+When `Core` receives one of the above mentioned events it will go and ask the
+wizards if they can handle the element and action.
+It will call:
+- `CanInspect` if the event is `oscd-wizard-inspection-request` 
+- `CanCreate` if the event is `oscd-wizard-creation-request`.
+
+##### Initiation
+
+In case of inspection the wizard would get the element. 
+```html
+<oscd-wizard-type-templates .element={element} .stuff={stuff}></oscd-wizard-type-templates>
+```
+
+If we want to create a new element it will get the parent element and the new tag name it has to create. Something like this:
+```html
+<oscd-wizard-type-templates .parentElement={parentElement} tag-name="DO" .stuff={stuff}></oscd-wizard-type-templates>
+```
+
+
+##### Wizarding Process
+
+This part happens entirely in the wizard plugin. 
+It can modify the XML document or just display it in some user-friendly way.
+
+##### Self Closing
+
+The wizard decides when it is finished and closes itself. 
+
+`Core` has always the possibility to close the wizard. 
+For example when clicking on the backdrop of a dialog.
+
+----
+
+## Considered Options
+
+### Define handled tag names in the plugin configuration
+
+An alternative to the `CanHandle` static function is to define the
+tag names in the plugin description e.g: in a string array called `handledTags`.
+
+```ts
+type WizardPlugin = Plugin & {
+    handledTags: string[];
+}
+```
+
+The advantage is that we can enforce their definition,
+what we cannot do with a static functions as of now.
+The disadvantages is that we cannot define the tags dynamically
+Another neutral difference is that the handled tags definition is at
+the registration point and not in the plugin's static function.
+So it can happen that the plugin changes which tags it can handle
+but we forget to change its configuration.
+
+
+### Alternatives of Requests
+
+#### [wizard-event.ts  ↗](https://github.com/ca-d/open-scd-core/blob/add-wizards/foundation/wizard-event.ts) by Christian Dinkle
+In this definition there is only one event (`oscd-wizard-request`)
+and two different event details (`CreateRequest` and `EditRequest`).
+This way the plugins need to determine themselves which
+event they got by examining each request.
+With two different events plugins can listen on one or both and can be sure
+which event they get without examining the event itself.
+Plugin developers may forget that they can get two different types of event details,
+but if we have two different events with always the same type of payload,
+there is nothing to forget.
```

Comment:
Do you mean forgetting in case wizard plugin developers do not look at the API and also choose not to use TypeScript? How likely do we think that case is? Do we want to design our API around it?

Do you also see the advantage of more straightforward backwards-compatible API growth in the future with this approach?